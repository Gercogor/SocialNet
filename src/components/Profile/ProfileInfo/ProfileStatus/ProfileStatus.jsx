import React, {useState} from 'react';
import styles from "./ProfileStatus.module.css";

const ProfileStatus = (props) => {

    let [editMode, setEditMode] = useState(false)
    let [newStatus, setNewStatus] = useState(props.status)

    const handleChange = () => {
        if (props.canChangeStatus===true) setEditMode(true)
    }

    const changeStatus = (newStatus) => {
        if (newStatus===props.status) return setEditMode(false)
        props.updateStatus(newStatus)
        setEditMode(false)
    }

    return (
        <div className={styles.status}>

            {
                editMode
                // true
                ?<input
                        autoFocus={true}
                        onBlur={(e)=>{
                            changeStatus(e.target.value)
                        }}
                        type="text"
                        onChange={(e)=>setNewStatus(e.target.value)}
                        value={newStatus}
                />
                :<span
                        onDoubleClick={handleChange}
                >
                        { props.status
                          ? props.status
                          : "no status"
                        }
                </span>
            }

        </div>
    );
};

export default ProfileStatus;