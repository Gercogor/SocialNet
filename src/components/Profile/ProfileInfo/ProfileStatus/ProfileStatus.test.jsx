import React from "react";
import {create} from 'react-test-renderer';
import ProfileStatus from "./ProfileStatus";
import {render, screen, fireEvent} from "@testing-library/react";


describe('ProfileStatus component', () => {
    test('status from props should be in the state', async () => {
        const element = create(<ProfileStatus status={'Fuck'}/>)
        const span = await element.root.findByType("span");
        expect(span).not.toBeNull();
    })
    test('status from props should be in the state v2', async () => {
        const element = create(<ProfileStatus status={'Fuck'}/>)
        const span = await element.root.findByType("span");
        // eslint-disable-next-line testing-library/no-node-access
        expect(span.props.children).toBe('Fuck');
    })
    test('upper test by render', async () => {
        render(<ProfileStatus status={'Fuck'}/>);
        expect(await screen.findByText('Fuck')).toBeDefined()
    })
    test('input should be displayed instead span',  () => {
        const element = create(<ProfileStatus canChangeStatus={true} status={'Fuck'}/>)
        // eslint-disable-next-line testing-library/await-async-query
        const span = element.root.findByType("span");
        span.props.onDoubleClick();
        // eslint-disable-next-line testing-library/await-async-query
        const input = element.root.findByType("input");
        // eslint-disable-next-line testing-library/no-node-access
        expect(input.props.value).toBe('Fuck');
    })

})


