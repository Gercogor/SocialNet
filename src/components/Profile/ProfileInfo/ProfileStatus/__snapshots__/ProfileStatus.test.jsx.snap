// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatus component input should be displayed instead span 1`] = `
<div
  className="status"
>
  <span
    onDoubleClick={[Function]}
  >
    Fuck
  </span>
</div>
`;

exports[`ProfileStatus component status from props should be in the state 1`] = `
<div
  className="status"
>
  <span
    onDoubleClick={[Function]}
  >
    Fuck
  </span>
</div>
`;
